tests/api/test_cluster.cpp

```diff
@@ -0,0 +1,79 @@
+
+#include <gtest/gtest.h>
+#include "fmt/xchar.h"
+
+#include <algorithm>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include "tests/test_utils/generate_cluster_desc.hpp"
+
+// TODO: change to tt_cluster
+#include "device/tt_device.h"
+#include "device/tt_cluster_descriptor.h"
+
+// TODO: do proper renaming.
+using Cluster = tt_SiliconDevice;
+
+// These tests are intended to be run with the same code on all kinds of systems:
+// E75, E150, E300
+// N150. N300
+// Galaxy
+
+// This test should be one line only.
+TEST(ApiTest, OpenAllChips) {
+
+    // TODO: This should not be needed. And could be part of the cluster descriptor probably.
+    // Note that cluster descriptor holds logical ids of chips.
+    // Which are different than physical PCI ids, which are /dev/tenstorrent/N ones.
+    // You have to see if physical PCIe is GS before constructing a cluster descriptor.
+    std::vector<int> pci_device_ids = PCIDevice::enumerate_devices();
+    std::set<int> pci_device_ids_set (pci_device_ids.begin(), pci_device_ids.end());
+
+    tt::ARCH device_arch = tt::ARCH::GRAYSKULL;
+    if (!pci_device_ids.empty()) {
+        // TODO: This should be removed from the API, the driver itself should do it.
+        int physical_device_id = pci_device_ids[0];
+        // TODO: remove logical_device_id
+        PCIDevice pci_device (physical_device_id, 0);
+        tt::ARCH device_arch = pci_device.get_arch();
```

Comment:
Not sure I understand, why are we redefining device_arch in this scope?

Reply:
It's an error, good catch!